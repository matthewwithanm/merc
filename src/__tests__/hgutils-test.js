/*
 * Copyright (c) 2016-present, Facebook, Inc.
 * All rights reserved.
 * @flow
 * @format
 */

/* eslint-env jest */

import path from 'path';
import {
  getRepoRoot,
  getCurrentRevisionHash,
  getMergeBaseHash,
  _getSubtreeCommitList,
} from '../hgutils';

describe('getRepoRoot', () => {
  test('it reports the root', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo1');
    const dir = path.join(repoRoot, 'a/b');
    const actual = await getRepoRoot(dir).toPromise();
    expect(actual).toBe(repoRoot);
  });

  test("it returns null when there's no repo", async () => {
    const dir = path.resolve(__dirname, './fixtures/not-a-repo/a/b');
    const actual = await getRepoRoot(dir).toPromise();
    expect(actual).toBe(null);
  });
});

describe('getCurrentRevisionHash', () => {
  test('it returns the current revision hash', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo2');
    const hash = await getCurrentRevisionHash(repoRoot).toPromise();
    expect(hash).toBe('19e0c8a7e1c0ff0099e0c642ea99cdba36e36923');
  });
});

describe('getMergeBaseHash', () => {
  test('it returns the current revision hash', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo2');
    const hash = await getMergeBaseHash(repoRoot).toPromise();
    expect(hash).toBe('c42dae43c973a141cf0f15d9bcb6f32093e1c1d2');
  });
});

describe('_getSubtreeCommitList', () => {
  test('it returns a list of commits with metadata', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo2');
    const list = await _getSubtreeCommitList(repoRoot).toPromise();
    expect(list).toMatchSnapshot();
  });
});
