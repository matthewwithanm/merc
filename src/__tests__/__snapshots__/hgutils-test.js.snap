// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_getSubtreeCommitList it returns a list of commits with metadata 1`] = `
"----node
47b65c76f24ae1897b7f54dca6a2d210caf78ec5
----p1node
c42dae43c973a141cf0f15d9bcb6f32093e1c1d2
----current
----phase
draft
----file_adds
----file_copies
----file_dels
----file_mods
files/file.txt
----node
6cc8d78c03a1ec393229a4d93fd013ca68baf2d7
----p1node
47b65c76f24ae1897b7f54dca6a2d210caf78ec5
----current
----phase
draft
----file_adds
----file_copies
----file_dels
----file_mods
files/file.txt
----node
19e0c8a7e1c0ff0099e0c642ea99cdba36e36923
----p1node
47b65c76f24ae1897b7f54dca6a2d210caf78ec5
----current
1
----phase
draft
----file_adds
----file_copies
----file_dels
----file_mods
files/file.txt
----node
260b695f01aa8d351bcda32830c174d8550cbfa1
----p1node
19e0c8a7e1c0ff0099e0c642ea99cdba36e36923
----current
----phase
draft
----file_adds
----file_copies
----file_dels
----file_mods
files/file.txt
----node
c42dae43c973a141cf0f15d9bcb6f32093e1c1d2
----p1node
bf2cf6e87a72ec7062e38e695bfb1151a4f63a9e
----current
----phase
public
----file_adds
files/file.txt
----file_copies
----file_dels
----file_mods
"
`;

exports[`_parseSubtreeCommitList it returns a set of commits 1`] = `
Set {
  Object {
    "addedFiles": Set {},
    "copiedFiles": Set {},
    "deletedFiles": Set {},
    "hash": "47b65c76f24ae1897b7f54dca6a2d210caf78ec5",
    "isCurrentRevision": false,
    "modifiedFiles": Set {
      "files/file.txt",
    },
    "parentHash": "c42dae43c973a141cf0f15d9bcb6f32093e1c1d2",
    "phase": "draft",
  },
  Object {
    "addedFiles": Set {},
    "copiedFiles": Set {},
    "deletedFiles": Set {},
    "hash": "6cc8d78c03a1ec393229a4d93fd013ca68baf2d7",
    "isCurrentRevision": false,
    "modifiedFiles": Set {
      "files/file.txt",
    },
    "parentHash": "47b65c76f24ae1897b7f54dca6a2d210caf78ec5",
    "phase": "draft",
  },
  Object {
    "addedFiles": Set {},
    "copiedFiles": Set {},
    "deletedFiles": Set {},
    "hash": "19e0c8a7e1c0ff0099e0c642ea99cdba36e36923",
    "isCurrentRevision": true,
    "modifiedFiles": Set {
      "files/file.txt",
    },
    "parentHash": "47b65c76f24ae1897b7f54dca6a2d210caf78ec5",
    "phase": "draft",
  },
  Object {
    "addedFiles": Set {},
    "copiedFiles": Set {},
    "deletedFiles": Set {},
    "hash": "260b695f01aa8d351bcda32830c174d8550cbfa1",
    "isCurrentRevision": false,
    "modifiedFiles": Set {
      "files/file.txt",
    },
    "parentHash": "19e0c8a7e1c0ff0099e0c642ea99cdba36e36923",
    "phase": "draft",
  },
  Object {
    "addedFiles": Set {
      "files/file.txt",
    },
    "copiedFiles": Set {},
    "deletedFiles": Set {},
    "hash": "c42dae43c973a141cf0f15d9bcb6f32093e1c1d2",
    "isCurrentRevision": false,
    "modifiedFiles": Set {},
    "parentHash": "bf2cf6e87a72ec7062e38e695bfb1151a4f63a9e",
    "phase": "public",
  },
}
`;

exports[`getCurrentRevisionHash it returns the current revision hash 1`] = `"19e0c8a7e1c0ff0099e0c642ea99cdba36e36923"`;

exports[`getMergeBaseHash it returns the current revision hash 1`] = `"c42dae43c973a141cf0f15d9bcb6f32093e1c1d2"`;
